src/NuGet.Services.EndToEnd/LicenseTests.cs

```diff
@@ -58,6 +60,27 @@ public async Task PushedPackageWithLicenseFileIsAvailableInV3()
             await _clients.FlatContainer.WaitForPackageAsync(package.Id, package.NormalizedVersion, _logger);
             var licenseFileList = await _clients.FlatContainer.TryAndGetFileStringContent(package.Id, package.NormalizedVersion, FlatContainerStringFileType.License, _logger);
             Assert.All(licenseFileList, x => Assert.Equal(package.Properties.LicenseFileContent, x));
+            VerifyLicenseUrlForV2V3Search(package, expectedPath);
+        }
+
+        private async void VerifyLicenseUrlForV2V3Search(Package package, string expectedPath)
+        {
+            await _clients.V2V3Search.WaitForPackageAsync(package.Id, package.FullVersion, _logger);
+            var searchServices = await _clients.V2V3Search.GetSearchServicesAsync(_logger);
+
+            foreach (var searchService in searchServices)
+            {
+                var results = await _clients.V2V3Search.QueryAsync(
+                    searchService,
+                    $"q=packageid:{package.Id}",
+                    _logger);
+
+                Assert.True(results.Data.Count > 0);
+                for (var i = 0; i < results.Data.Count; i++)
```

Comment:
There should be exactly one result, right?

Also, please assert package version to be confident in the result.